src/approaches/grammar_search_invention_approach.py

```diff
@@ -799,8 +808,11 @@ def _relaxation_h(atoms: Set[GroundAtom], depth: int = 0) -> float:
 class _ExactHeuristicBasedScoreFunction(_HeuristicBasedScoreFunction):  # pylint:disable=abstract-method
     """Implement _generate_heuristic() with task planning."""
 
+    heuristic_names: Sequence[str] = ("exact", )
```

Comment:
This would be a class variable instead of an instance variable right? Should it be `heuristic_names: Sequence[str] = field(default=("exact", ))`?

Reply:
and maybe also `init=False`?